dashboard/app/models/level_source.rb

```diff
@@ -124,4 +124,14 @@ def get_hint_from_source(source)
       get_hint_from_source_internal(including: source)
     end
   end
+
+  # Old flappy levels used a different block type as their when run. Migrate
+  # these as we try to access them
+  def replace_old_when_run_blocks()
+    if self.level.game.name == 'Flappy' and self.data.include?('flappy_whenRunButtonClick')
```

Comment:
the only selfs that are necessary are the ones on the left hand side of the assignments (to disambiguate between calling the instance methods and creating a new local variable by assignment).